common/crypto/attestation-api/evidence/verify-dcap-direct-evidence.cpp

```diff
@@ -0,0 +1,268 @@
+/*
+ * Copyright 2024 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <cstdio>
+#include <algorithm>
+#include <vector>
+#include <string>
+#include <array>
+#include "types/types.h"
+#include "types/hex_string.h"
+#include "error.h"
+#include "logging.h"
+#include "crypto/sha256.h"
+#include "crypto/verify_dcap_direct/get_dcap_certificate.h"
+#include "base64/base64.h"
+#include "attestation_tags.h"
+
+#include "sgx_quote_3.h"
+#include "SgxEcdsaAttestation/QuoteVerification.h"
+#include "QuoteGeneration/quote_wrapper/common/inc/sgx_ql_lib_common.h"
+#include "CertVerification/CertificateChain.h"
+
+// < JSON include
+#define JSON_HAS_CPP_11
+#define JSON_NO_IO 1
+#include <errno.h>
+#include "nlohmann/json.hpp"
+using json = nlohmann::json;
+// JSON include >
+
+#include "PckParser/CrlStore.h"
+#include "../../AttestationCommons/include/Utils/TimeUtils.h"
+
+/********************************************************
+ * Internal function to modify "in-place" the collateral
+ * data structure, so that pointers point to buffers
+ * appended to the structure
+ *******************************************************/
+void deserialize_collateral(uint8_t* p)
+{
+    sgx_ql_qve_collateral_t* c = (sgx_ql_qve_collateral_t*)p;
+
+    c->pck_crl_issuer_chain     = (char*)p + sizeof(sgx_ql_qve_collateral_t);
+    c->root_ca_crl              = (char*)c->pck_crl_issuer_chain + c->pck_crl_issuer_chain_size;
+    c->pck_crl                  = (char*)c->root_ca_crl + c->root_ca_crl_size;
+    c->tcb_info_issuer_chain    = (char*)c->pck_crl + c->pck_crl_size;
+    c->tcb_info                 = (char*)c->tcb_info_issuer_chain + c->tcb_info_issuer_chain_size;
+    c->qe_identity_issuer_chain = (char*)c->tcb_info + c->tcb_info_size;
+    c->qe_identity              = (char*)c->qe_identity_issuer_chain + c->qe_identity_issuer_chain_size;
+}
+
+
+bool verify_dcap_direct_evidence(ByteArray& evidence, ByteArray& expected_statement, ByteArray& expected_code_id)
+{
+    ByteArray quote;
+    ByteArray certification_data;
+    uint32_t certification_data_size;
+    uint16_t certification_data_type;
+    ByteArray collateral;
```

Comment:
These vars seem to be used only in verify block, so why not declare them there?  Makes it easier when reading code to see var definitions ...